examples/dpapi.py

```diff
@@ -503,6 +475,63 @@ def run(self):
                 # Just print the data
                 blob.dump()
 
+        elif self.options.action.upper() == 'CREDHIST':
+            fp = open(options.file, 'rb')
+            data = fp.read()
+            chf = CREDHIST_FILE(data)
+
+            keys = []
+
+            # Handle key options
+            if self.options.key and self.options.sid:
+                key = unhexlify(self.options.key[2:])
+                keys = deriveKeysFromUserkey(self.options.sid, key)
+
+            elif self.options.sid and self.options.key is None:
+                # Do we have a password?
+                if self.options.password is None:
+                    # Nope let's ask it
+                    from getpass import getpass
+                    password = getpass("Password:")
+                else:
+                    password = options.password
+                keys = deriveKeysFromUser(self.options.sid, password)
+
+            else:
+                chf.dump()
+                return
+
+            if self.options.entry is None:
+                # First find the correct key to the 1st entry
+                real_key = None
+                for k in keys:
+                    chf.decrypt_entry_by_index(0, k)
+                    if chf.credhist_entries_list[0].pwdhash is not None:
+                        real_key = k
+                        break
+
+                # Wrong key
+                if real_key is None:
+                    chf.dump()
+
+                else:
+                    chf.decrypt(real_key)
+                    chf.dump()
+
+                    # Fully successful decryption
+                    if chf.credhist_entries_list[-1].pwdhash is not None:
+                        return
+
+            else:
+                real_key = None
```

Comment:
The variable is not used
```suggestion
            else:
```